### PR TITLE
arch: imx6: Add the pgheap for imx6

### PR DESCRIPTION
**Pull request by masayuki2009**
## Summary

- This commit adds the pgheap for imx6

## Impact

- None

## Testing

- Tested with sabre-6quad:netknsh (not merged yet)

### CODE REVIEW

File: arch/arm/src/imx6/imx_pgalloc.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+/****************************************************************************
+ * Name: up_allocate_pgheap
+ *
+ * Description:
+ *   If there is a page allocator in the configuration, then this function
+ *   must be provided by the platform-specific code.  The OS initialization
+ *   logic will call this function early in the initialization sequence to
+ *   get the page heap information needed to configure the page allocator.
+ *
```
Comment (acassis) [inline arch/arm/src/imx6/imx_pgalloc.c:91]: Please include "Input Parameters" and "Returned Values"

File: arch/arm/src/imx6/imx_pgalloc.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  *heap_size  = CONFIG_IMX6_DDRCS_PGHEAP_SIZE;
+}
+
+/****************************************************************************
+ * Name: imx_virtpgaddr
+ *
+ * Description:
+ *   Check if the physical address lies in the page pool and, if so
+ *   get the mapping to the virtual address in the user data area.
+ *
```
Comment (acassis) [inline arch/arm/src/imx6/imx_pgalloc.c:109]: Ditto

File: arch/arm/src/imx6/imx_pgalloc.h [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: imx_physpgaddr
+ *
+ * Description:
+ *   Check if the virtual address lies in the user data area and, if so
+ *   get the mapping to the physical address in the page pool.
+ *
```
Comment (acassis) [inline arch/arm/src/imx6/imx_pgalloc.h:62]: Ditto

File: arch/arm/src/imx6/imx_pgalloc.h [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+#define imx_physpgaddr(vaddr) arm_physpgaddr(vaddr)
+
+/****************************************************************************
+ * Name: imx_virtpgaddr
+ *
+ * Description:
+ *   Check if the physical address lies in the page pool and, if so
+ *   get the mapping to the virtual address in the user data area.
+ *
```
Comment (acassis) [inline arch/arm/src/imx6/imx_pgalloc.h:73]: Ditto

**Comment by masayuki2009** [comment]
@acassis 

I added comments and removed unused functions.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**